### PR TITLE
Create Rock, Paper, Shotgun.js

### PR DESCRIPTION
**Pull request by owcz**
translator for [Rock, Paper, Shotgun](http://rockpapershotgun.com), written in Framework

### CODE REVIEW

**Comment by adam3smith** [comment]
something went wrong here, possibly related to the rebase. You don't want to add the Polygon translator again.

File: Rock, Paper, Shotgun.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+FW.Scraper({
+itemType         : 'blogPost',
+detect           : FW.Url().match(/\/(\d{4})\/(\d{2})\/(\d{2})\//),
+title            : FW.Xpath('//header/h2/a').text().trim(),
+attachments      : [{ url: FW.Url(),
+  title:  "Rock, Paper, Shotgun snapshot",
+  type: "text/html" }],
+creators         : FW.Xpath('//header/aside/p/a').text().cleanAuthor("author"),
+date             : FW.Url().match(/(\d{4})\/(\d{2})\/(\d{2})/).replace(/\//g,"-"),
+tags             : FW.Xpath('//footer/p[@class="tags"]').text(),
```
Comment (adam3smith) [inline Rock, Paper, Shotgun.js]: `.split(/,\s*/)` it would seem from looking at the tests, right?

File: Rock, Paper, Shotgun.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) { return FW.detectWeb(doc, url); }
+function doWeb(doc, url) { return FW.doWeb(doc, url); }
+
+FW.Scraper({
+itemType         : 'blogPost',
+detect           : FW.Url().match(/\/(\d{4})\/(\d{2})\/(\d{2})\//),
```
Comment (adam3smith) [inline Rock, Paper, Shotgun.js]: why the parentheses here and below? You're not using the different matched elements, so might as well leave them out (and gain some admittedly minimal efficiency)

**Comment by adam3smith** [comment]
For the multi-scrapers (or really for all scrapers) you can simply have several of them. See e.g. https://github.com/zotero/translators/blob/master/Wikimedia%20Commons.js for a complex version of a FW translator. i.e. in your case

```
FW.MultiScraper({
itemType         : 'multiple',
detect           : FW.Url().match(/\/category\//), // category pages
choices          : {
  titles :  FW.Xpath('//div[@class="post-inner"]/h2/a').text().trim(),
  urls    :  FW.Xpath('//div[@class="post-inner"]/h2/a').key("href")
}
});

FW.MultiScraper({
itemType         : 'multiple',
detect           : FW.Url().match(/\/\?s\=/), // search pages
choices          : {
  titles :  FW.Xpath('//li/div/h3/a').text().trim(),
  urls    :  FW.Xpath('//li/div/h3/a').key("href")
}
});
```

They then detect sequentially (i.e. the first one to detect fires, any later ones are ignored).

**Comment by adam3smith** [comment]
That should be all. Thanks!

**Comment by owcz** [comment]
Squashed 🍆

**Comment by adam3smith** [comment]
That was faster. Nice work & thanks.

**Comment by owcz** [comment]
@adam3smith: What's the easiest way to have a translator (such as this one) default to EM for all fields not specified? It looks like EM can do almost all of the work for this translator besides for the author field. (I'm going to email the pub about supporting full metadata but thought I should know for the future.)

**Comment by adam3smith** [comment]
https://github.com/zotero/translators/blob/master/Le%20Monde.js
is a good example. In other words, call the EM translator, and then fix everything you need to in the item handler, where you can modify what's already in the item as well as add additional fields.

More generally, @zuphilip has collected a very good set of sample codes and translators for specific situations here: https://github.com/zuphilip/translators/wiki 
I very much recommend both his code blocks and the list of common mistakes.

**Comment by owcz** [comment]
Nice. Thanks! So just to clarify, there's no EM fallback option baked into Z Framework?

**Comment by adam3smith** [comment]
correct, you have to explicitly call it.

**Pull request merged**